File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    
    // 방문 배열
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        visited = new boolean[n + 1];
        
        for (int i = 1; i <= n; i++) {
            if (!visited[i]) {                
                DFS(i, n, computers);
                answer++;
            }
        }
        
        return answer;
    }
    
    private static void DFS(int c, int n, int[][] computers) {
        
        visited[c] = true;
        
        for (int i = 1; i <= n; i++) {
            if (!visited[i] && computers[c - 1][i - 1] == 1) {
                DFS(i, n, computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = Array.from({length : n}, () => false)
    
    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            dfs(i)
            answer += 1
        }
    }
    
    function dfs(computer_idx) {
        visited[computer_idx] = true        
        for (let c = 0; c < n; c++) {
            if (!visited[c] && computers[computer_idx][c] == 1) {
                dfs(c)
            }
        }        
    }    
    
    return answer;
}6



